### ISSUE TITLE
[BUG] The jar has problems after decompilation.

### ISSUE THREAD

**Issue by zhongqingsong** [ThexXTURBOXx/dex2jar#27]
Decompiled an apk to get a jar. After running, it was found to crash all the time. Finally, it was found that the decompiled app jar error.
You can see "-I" changed to "_I". It's not right. Below show:
![image](https://user-images.githubusercontent.com/35858818/197459580-f32e8808-ed96-4459-8d2b-8d5b0793cbb2.png)

The origin code:
![image](https://user-images.githubusercontent.com/35858818/197460225-0091c006-7de8-4d19-ba02-0de2a1dfa593.png)

I know, we can change smali file to fix. But let me just mention that. It would be nice to have a tool that can fix that.

**Comment by ThexXTURBOXx** [comment]
Thanks for reporting that! Could you please share the original APK/DEX file?

**Comment by zhongqingsong** [comment]
[dexs.zip](https://github.com/ThexXTURBOXx/dex2jar/files/9878160/dexs.zip)


> Thanks for reporting that! Could you please share the original APK/DEX file?
hello, this is dex file. APK too big, so I zip dex.

**Comment by ThexXTURBOXx** [comment]
Thanks for sharing. I will see what I can do about it soon

**Comment by Syer10** [comment]
@ThexXTURBOXx My application has also starting receiving this issue. We have a lot of users so a fix would be appreciated.
Actual method: `public static final long kotlin.time.Duration.getInWholeMilliseconds-impl(long)`
Error:
```java
java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: 'long kotlin.time.Duration.getInWholeMilliseconds_impl(long)'
Caused by: java.lang.NoSuchMethodError: 'long kotlin.time.Duration.getInWholeMilliseconds_impl(long)'
	at eu.kanade.tachiyomi.extension.all.mangadex.MDConstants.<clinit>(Unknown Source)
	at eu.kanade.tachiyomi.extension.all.mangadex.MangaDex.sanitizeExistingUuidPrefs(Unknown Source)
	at eu.kanade.tachiyomi.extension.all.mangadex.MangaDex.<init>(Unknown Source)
	at eu.kanade.tachiyomi.extension.all.mangadex.MangaDex.<init>(Unknown Source)
	at eu.kanade.tachiyomi.extension.all.mangadex.MangaDexEnglish.<init>(Unknown Source)
	at eu.kanade.tachiyomi.extension.all.mangadex.MangaDexFactory.createSources(Unknown Source)
	at suwayomi.tachidesk.manga.impl.extension.Extension.installAPK(Extension.kt:161)
	at suwayomi.tachidesk.manga.impl.extension.Extension$installAPK$1.invokeSuspend(Extension.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	... 7 common frames omitted
```

File: [tachiyomi-all.mangadex-v1.4.189.zip](https://github.com/ThexXTURBOXx/dex2jar/files/12910973/tachiyomi-all.mangadex-v1.4.189.zip)

**Comment by ThexXTURBOXx** [comment]
I thought I had fixed that already. Are you using the newest version of my fork? Also, can you share an example APK/DEX file? Perhaps if needed, you can also share it via my private email

**Comment by Syer10** [comment]
We are on v74, someone did some testing and its happening with every version above v64, I shared the apk above. https://github.com/ThexXTURBOXx/dex2jar/files/12910973/tachiyomi-all.mangadex-v1.4.189.zip

**Comment by ThexXTURBOXx** [comment]
Oh sorry, did not see that first!
You said it does not work with every version above v64. Have you tried versions below (and including) v64? I would imagine it also does not work there either, right?

**Comment by Syer10** [comment]
It works fine on v64, but every version after it has this issue

**Comment by ThexXTURBOXx** [comment]
I see what is the problem: The [new `fixName`](https://github.com/ThexXTURBOXx/dex2jar/compare/v64...v65#diff-c4ca3aae346c8f9fe99b4f9dea98c44e4915deba357b498b615c54f90084e755R15-R20) function rigorously renames functions without checking for their existence. Since the function you are referring to is built-in, we should *not* replace it.
Not an easy fix, but I will hopefully be able to take a look at it soon

**Comment by Syer10** [comment]
I am going to revert to v64 for now, I'll keep watching this issue for updates

**Comment by ThexXTURBOXx** [comment]
I have deployed a temporary fix for this issue which is available as part of 2.4.10.
If you need a fix, you need to set the parameter `dont sanitize names`.
However, be aware that this could lead to other incorrect behaviour regarding lambda expressions.
A proper fix is probably impossible, but this will do for now.
